src/celeritas/em/distribution/UrbanMscScatter.hh

```diff
@@ -158,9 +158,11 @@ UrbanMscScatter::UrbanMscScatter(const UrbanMscRef&       shared,
 
     lambda_ = helper_.msc_mfp(inc_energy_);
 
-    // Convert the geometry path length to the true path length
-    true_path_ = this->calc_true_path(
-        input_.true_path, input_.geom_path, input_.alpha);
+    // Convert the geometry path length to the true path length, but do not
+    // recalculate the true path if the step is not limited by geometry
+    true_path_ = (input_.geo_limited) ? this->calc_true_path(
+                     input_.true_path, input_.geom_path, input_.alpha)
+                                      : input_.true_path;
```

Comment:
I see that we're using `true_path_` here (which is only `input_.true_path` if not geo-limited) and `input_.true_path` in other places in MscScatter. Can you take the opportunity to remove `input` from the member data, keeping only the individual components that are used inside `operator()`, and documenting the difference between `true_path` and `input.true_path`?

Reply:
Nice a catch! Actually input_.true_path should be only used in the constructor, so the one in the operator should be true_path_, which is fixed now.   Regarding to input_, do you want to introduce three additional data members, limit_min_, is_displace_ and geom_path_ by replacing input_?

Reply:
I think that is a good idea: locally store only what we're using.